src/Honeycomb.OpenTelemetry/TracerProviderBuilderExtensions.cs

```diff
@@ -45,23 +45,16 @@ public static TracerProviderBuilder AddHoneycomb(this TracerProviderBuilder buil
             {
                 throw new ArgumentNullException(nameof(options), "No Honeycomb options have been set in appsettings.json, environment variables, or the command line.");
             }
-            
-            if (string.IsNullOrWhiteSpace(options.TracesApiKey))
-                Console.WriteLine("WARN: missing traces API key");
-            if (string.IsNullOrWhiteSpace(options.TracesDataset))
-                Console.WriteLine("WARN: missing traces dataset");
+
+            // if serviceName is null, warn and set to default
+            if (string.IsNullOrWhiteSpace(options.ServiceName)) {

```

Comment:
Should this only warn when service name is empty and the API key is not legacy?

```suggestion
            if (!options.IsLegacyKey() && string.IsNullOrWhiteSpace(options.ServiceName)) {
```

Reply:
It's true we'll continue to send to dataset for a legacy key and don't technically need service name, but thought it would still be useful to know if service name is empty. Especially when it comes time to migrate later, folks might want to know ahead of time that they are missing a service name and fix that proactively.